### PR TITLE
Updates vend-a-trays to have a unique UI, improves accessibility. 

### PR DESCRIPTION
**Pull request by ArcaneMusic**
Might as well maintain my shitcode

## About The Pull Request

When I did this PR originally, I did everything in my power to avoid learning how to make a UI, now that I'm older, wiser, smarter, and my brain is scarred with trauma, this will replace the vend-a-tray's ID card interactions with a UI, showcasing the object inside, alongside controls to buy the object, open the object, or change the cost. To sell anything, the Vend-A-Tray has to be registered, which is done with a single click of the button now, and the owner is shown off for all to see.

The existing, mapped vend-a-trays have been replaced with a kitchen specific subtype which requires kitchen access, whereas the new default type has no access requirements to use. Additionally, it may now be constructed through a new service machine circuit board, which uses the same parts to construct, but can now be constructed with relative ease and considerably less headaches.
image

## Why It's Good For The Game
I think more people will use this feature with a clean and concise UI, and it gives me a bit of time to figure out how to work out an image component in for other TGUIs to use.

## Changelog
:cl:
feature: Vend-A-Trays can now be produced with a machine circuitboard instead of a display case frame.
tweak: Vend-A-Trays are now operated through a UI.
/:cl:

### CODE REVIEW

**Comment by Angustmeta** [comment]
Yeah never saw these used. 

Probably fixes #50722.

**Review by stylemistake — DISMISSED** [review]

**Review by stylemistake — DISMISSED** [review]

File: code/game/objects/structures/displaycase.dm [hunk 1]
```diff
@@ -396,64 +396,135 @@
...
+				to_chat(usr, "<span class='notice'>[potential_acc] has no account registered!</span>")
+				return
+			if(!account.has_money(sale_price))
+				to_chat(usr, "<span class='notice'>You do not possess the funds to purchase this.</span>")
+				return TRUE
+			else
+				account.adjust_money(-sale_price)
+				if(payments_acc)
+					payments_acc.adjust_money(sale_price)
+				usr.put_in_hands(showpiece)
```
Comment (ShizCalev) [inline code/game/objects/structures/displaycase.dm:453]: what if they're using tk

Comment (ArcaneMusic) [reply code/game/objects/structures/displaycase.dm:453]: That's checked on 437 in the same instance, canUseTopic with NO_TK.

File: code/game/objects/structures/displaycase.dm [hunk 2]
```diff
@@ -396,64 +396,135 @@
...
+
+/obj/structure/displaycase/forsale/ui_data(mob/user)
+	var/list/data = list()
+	var/register = FALSE
+	if(payments_acc)
+		register = TRUE
+		data["owner_name"] = payments_acc.account_holder
+	if(showpiece)
+		data["product_name"] = capitalize(showpiece.name)
+		var/base64 = icon2base64(icon(showpiece.icon, showpiece.icon_state))
```
Comment (AnturK) [inline code/game/objects/structures/displaycase.dm:414]: This is expensive. Very bad idea to add this in ui_data of processing UI. Needs a cache and ideally making this static/ui not auto processing.

File: code/game/objects/structures/displaycase.dm [hunk 3]
```diff
@@ -396,64 +396,135 @@
...
+			if(!showpiece)
+				to_chat(usr, "<span class='notice'>There's nothing for sale.</span>")
+				return TRUE
+			if(broken)
+				to_chat(usr, "<span class='notice'>[src] appears to be broken.</span>")
+				return TRUE
+			if(!payments_acc)
+				to_chat(usr, "<span class='notice'>[src] hasn't been registered yet.</span>")
+				return TRUE
+			var/confirm = alert(usr, "Purchase [showpiece] for [sale_price]?", "Purchase?", "Confirm", "Cancel")
```
Comment (AnturK) [inline code/game/objects/structures/displaycase.dm:436]: You could use the tgui confirm button here instead of alert.

**Review by AnturK — CHANGES_REQUESTED** [review]

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Linter says no

**Pull request merged**